File: HW2- Tetris2 - AutoPlay/200104004099_EmreOytun/driver2.cpp
#include <iostream>
#include "Tetromino.h"
#include "Tetris.h"
#include <vector>
#include <stdlib.h>
#include <time.h>
using namespace std;

Tetromino randomTetromino() {
	vector<Tetromino_t> typeList {Tetromino_t::I, Tetromino_t::O, Tetromino_t::L, Tetromino_t::S, Tetromino_t::Z, Tetromino_t::J, Tetromino_t::T};
	int randomNum = (int)rand() % 7; 
	
	return Tetromino(typeList.at(randomNum));
}

int main() {
	srand(time(NULL));
	
	cout << "\t\t ********* DRIVER - 2 ******** \n\n";
	
	int len = 0;
	int width = 0;
	
	
	/* Get the length of board. */
	bool isRead = false;
	do {
		cout << "Enter board length : ";
		cin >> len;
	
		/* Check the validity of the input. */
		if (cin.fail()) {
			cout << "Enter a valid input!\n";
			cin.clear();
			cin.ignore(256, '\n');	// Ignore the stream.
		}
		else if (len<0) cout << "Enter a positive number!\n";
		else isRead = true;
				
	}while (!isRead);
	
	/* Get the width of board. */
	isRead = false;
	do {
		cout << "Enter board width : ";
		cin >> width;
	
		/* Check the validity of the input. */
		if (cin.fail()) {
			cout << "Enter a valid input!\n";
			cin.clear();
			cin.ignore(256, '\n');	// Ignore the stream.
		}
		else if (width<0) cout << "Enter a positive number!\n";
		else isRead = true;
				
	}while (!isRead);
	
	/* Initialize a Tetris instance with given board parameters. */
	Tetris tetris(len, width);
	
	bool isDone = false;
	while(!isDone) {
		isRead = true;
		char ch;
		cout << "Enter a tetromino (R - Random) : ";
		cin >> ch;
		vector<Tetromino> tlist;
		
		switch (ch) {
			
				case 'I':
					tlist.push_back(Tetromino(Tetromino_t::I));
					break;
				
				case 'O' :
					tlist.push_back(Tetromino(Tetromino_t::O));
				 	break;
				 	
				case 'T' :
				 	tlist.push_back(Tetromino(Tetromino_t::T));
			 		break;
				 	
			 	case 'J' :
				 	tlist.push_back(Tetromino(Tetromino_t::J));
				 	break;
				
				case 'L' :
					tlist.push_back(Tetromino(Tetromino_t::L));
					break;
					
				case 'Z' :
					tlist.push_back(Tetromino(Tetromino_t::Z));
					break;
					
				case 'S' :
					tlist.push_back(Tetromino(Tetromino_t::S));
					break; 	
				
				case 'R' :
					tlist.push_back(randomTetromino());
					break;
				
				default :
					cout << "Enter a valid input!\n";
					cin.clear();
					cin.ignore(256, '\n');
					isRead = false;
					break;
				 
			}
			
			if (isRead) {
				Tetromino curTet = tlist[tlist.size()-1];
				
				tetris.add(curTet);
				
				if (tetris.gameStatus() == false) {
					cout << "Current tetromino : \n";
					curTet.print();
					cout << "\n";
					
					cout << "Current tetromino cannot be added at the top row in the middle !\n\n";
					
					cout << "The board : \n";
					tetris.draw();
				
					cout << "\nGAME OVER!\n";
					
					isDone = true;
				}
				
				else {
					tetris.animate();
				}
				
				cout << "\n";	
			}
	}
	
	return 0;
}

File: HW7- Tetris in Java/Driver2.java
import java.util.Random;
import java.util.Scanner;

public class Driver2 {
    
    public static Tetromino randomTetromino() {
        TetrominoType[] typeList = new TetrominoType[] {TetrominoType.I, TetrominoType.O, TetrominoType.L, TetrominoType.S, TetrominoType.Z, TetrominoType.J, TetrominoType.T};
        
        Random random = new Random();
        int randomNum = random.nextInt(7);
        randomNum = randomNum % 7; 

        return new Tetromino(typeList[randomNum]);
    }
    

    public static void main(String[] args) {
        System.out.println("\t\t ********* DRIVER - 2 ******** \n\n");
	
        int len = 0;
        int width = 0;
        
        Scanner scanner = new Scanner(System.in);
        
        /* Get the length of board. */
        boolean isRead = false;
        do {
            len = -1;
            System.out.println("Enter board length : ");
            
            /* Check the validity of the input. */
            if (!scanner.hasNextInt()) {
                System.out.println("Enter a valid input!");
                scanner.nextLine();
            }
            else {
                len = scanner.nextInt();
                if (len<0) System.out.println("Enter a positive number!");
                else isRead = true;
            }
        
        }while (!isRead);
        
        /* Get the width of board. */
        isRead = false;
        do {
            width = -1;
            System.out.println("Enter board width : ");
        
            /* Check the validity of the input. */
            if (!scanner.hasNextInt()) {
                System.out.println("Enter a valid input!");
                scanner.nextLine();
            }
            else {
                width = scanner.nextInt();
                if (width<0) System.out.println("Enter a positive number!");
                else isRead = true;
            }
    
        }while (!isRead);
        
        /* Initialize a Tetris instance with given board parameters. */
        Tetris tetris = new Tetris(len, width);
        
        scanner.nextLine();
        boolean isDone = false;
        while(!isDone) {
            isRead = true;
            char ch;
            System.out.println("Enter a tetromino (R - Random, 'T','O','L','S','J','Z','S', Q - Quit): ");
            
            String nextLine = scanner.nextLine();
            if (nextLine.length() == 0) {
                ch = 'E'; // 'E' for empty.
            }
            else {
                ch = nextLine.charAt(0);
            }
            Tetromino newT = null;
            
            switch (ch) {
                
                    case 'I' :
                    case 'i' :
                        newT = new Tetromino(TetrominoType.I);
                        break;
                    
                    case 'O' :
                    case 'o' :
                        newT = new Tetromino(TetrominoType.O);
                        break;
                        
                    case 'T' :
                    case 't' :
                        newT = new Tetromino(TetrominoType.T);  
                        break;
                        
                    case 'J' :
                    case 'j' :
                        newT = new Tetromino(TetrominoType.J);
                        break;
                    
                    case 'L' :
                    case 'l' :
                        newT = new Tetromino(TetrominoType.L);
                        break;
                        
                    case 'Z' :
                    case 'z' :
                        newT = new Tetromino(TetrominoType.Z);
                        break;
                        
                    case 'S' :
                    case 's' :
                        newT = new Tetromino(TetrominoType.S);
                        break; 	
                    
                    case 'R' :
                    case 'r' :
                        newT = randomTetromino();    
                        break;
                    
                    case 'Q':
                    case 'q':
                        System.out.println("Program is closing... Goodbye!");
                        isRead = false;
                        isDone = true;
                        break;

                    default :
                        System.out.println("Enter a valid input!\n");
                        isRead = false;
                        break;
                    
                }
                
                if (isRead) {
                    Tetromino curTet = newT;
                    
                    tetris.add(curTet);
                    
                    if (tetris.gameStatus() == false) {
                        System.out.println("Current tetromino : \n");
                        curTet.print();
                        System.out.println("\n");
                        
                        System.out.println("Current tetromino cannot be added at the top row in the middle !\n\n");
                        
                        System.out.println("The board : \n");
                        tetris.draw();
                    
                        System.out.println("\nGAME OVER!\n");
                        
                        isDone = true;
                    }
                    
                    else {
                        tetris.animate();
                    }
                    
                    System.out.println("\n");	
                }
        
        }
  
        scanner.close();
    }

}

